### PR TITLE
WELD-2259 Upgrade Weld API to 3.0.Alpha23

### PR DESCRIPTION
**Pull request by mkouba**
- CDI API 2.0.Alpha6
- javax.annotation 1.3

### CODE REVIEW

**Comment by mkouba** [comment]
Wait till CDI TCK 2.0.0.Alpha9 hits maven cenral repo.

File: jboss-tck-runner/src/test/tck20/tck-tests.xml [hunk 1]
```diff
@@ -92,6 +92,84 @@
...
+                </methods>
+            </class>
+
+            <!-- WELD-2269 -->
+            <class name="org.jboss.cdi.tck.tests.context.request.event.RequestScopeEventTest">
+                <methods>
+                    <exclude name=".*"/>
+                </methods>
+            </class>
+
```
Comment (tremes) [inline jboss-tck-runner/src/test/tck20/tck-tests.xml]: I think it's missing SE tests. I can see following failures locally:
- InvalidContextualReferenceTest
- ContextSETest
- ActivateRequestContextByInterceptorTest
- ActivateRequestContextProgrammaticallyTest.

Comment (mkouba) [inline jboss-tck-runner/src/test/tck20/tck-tests.xml]: Oops, you're right.

**Comment by mkouba** [comment]
Merged!

**Pull request closed without merging**